### PR TITLE
Add error reporting to web-worker.js

### LINKED ISSUE #7453: Web worker error reporting
We can slightly generify the code in `src/service-worker/error-reporting.js`.

### PR DESCRIPTION
**Pull request by choumx**
Fixes #7453.

- Generic-ify `src/service-worker/error-reporting.js` 
- Add error reporting to `web-worker.js`
- Verified error URLs are correct locally

/to @jridgewell @kmh287

### CODE REVIEW

File: src/worker-error-reporting.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
+/**
```
Comment (jridgewell) [inline src/worker-error-reporting.js]: Can we move this into a common directory? Ie, `/worker`, with `/worker/service-worker` and `/worker/web-worker`?

File: build-system/tasks/presubmit-checks.js [hunk 2]
```diff
@@ -488,17 +488,25 @@ var forbiddenTerms = {
       'build-system/amp.extern.js',
       'build-system/tasks/prepend-global/test.js',
       'build-system/tasks/prepend-global/index.js',
+      'dist.3p/current/integration.js',
```
Comment (kmh287) [inline build-system/tasks/presubmit-checks.js]: Is this change just alphabetization? If so, how would you feel about alphabetizing the rest of the whitelist?

Comment (choumx) [inline build-system/tasks/presubmit-checks.js]: Done.

File: src/worker-error-reporting.js (hunk 1 above)
Comment (choumx) [inline src/worker-error-reporting.js]: Not sure adding another directory layer is worth it for a single file. I'll do this if we add more shared worker code.

**Pull request merged**